mono/metadata/guard-page.txt

```diff
@@ -0,0 +1,100 @@
+// Due to the guard page mechanism providing gradual commit of Windows stacks,
```

Comment:
I guess could put one or two two small functions here to justify its existance?
i.e. the line that calls max?

Reply:
You mean under the 4:th way of doing it and the way we use here? Yeah that would make sense.

Reply:
I mean, this single line:

```
aligned_stack_start = MAX (aligned_stack_start, info->client_info.info.windows_tib->StackLimit);
```


could be a function:

static gsize
mono_thread_aligned_stack_start (info)
{
 void *aligned_stack_start aligned_stack_start = (void*)(mword) ALIGN_TO ((mword)info->client_info.stack_start, SIZEOF_VOID_P);
#if _WIN32
 // Huge comment, single-intsanced
 aligned_stack_start  = (aligned_stack_start, info->client_info.info.windows_tib->StackLimit);
#endif
 return aligned_stack_start ;
}

Reply:
Not a big fan of having this huge comment with code samples included in source code, then I guess the way you have it now is OK (in separate file) or reduce the comment to smaller size (without source code example) and do as stated above.

Reply:
The comment was already large, and matter is very esoteric. I agree the comment is then really really large. I could blog it perhaps.. I'm about to put the comment in the code. But ok either way.

Reply:
i.e. very esoteric merits larger comment imho, but maybe I exceeded the limit.